From 123a03d7e2ae90da1b8a00ada136f439ba5672db Mon Sep 17 00:00:00 2001
From: Fumiya Kohzu <kohzu.fumiya@jp.panasonic.com>
Date: Mon, 20 Nov 2023 19:37:50 +0900
Subject: [PATCH] Remove card_index option support from virtio-loopback-driver,
 because the kernel needs to be patched to run that option in rvgpu-proxy.

Upstream-Status: Inappropriate [embedded specific]
---
 virtio_lo.h        | 1 -
 virtio_lo_device.c | 1 -
 virtio_lo_device.h | 1 -
 virtio_lo_driver.c | 3 ---
 4 files changed, 6 deletions(-)

diff --git a/virtio_lo.h b/virtio_lo.h
index a07fa91..0da681e 100644
--- a/virtio_lo.h
+++ b/virtio_lo.h
@@ -25,7 +25,6 @@ struct virtio_lo_devinfo {
 	__u64 features; /* IN/OUT */
 	__u32 config_size; /* IN */
 	__s32 config_kick; /* IN */
-	__s32 card_index; /* IN */
 	__u32 padding; /* IN */
 	__u8 *config; /* IN/OUT */
 	struct virtio_lo_qinfo *qinfo; /* IN/OUT */
diff --git a/virtio_lo_device.c b/virtio_lo_device.c
index 39e8788..22687b7 100644
--- a/virtio_lo_device.c
+++ b/virtio_lo_device.c
@@ -171,7 +171,6 @@ static long vilo_ioctl_adddev(struct virtio_lo_owner *owner,
 
 	dev->device_id = di.device_id;
 	dev->vendor_id = di.vendor_id;
-	dev->card_index = di.card_index;
 	dev->nqueues = di.nqueues;
 	dev->features = dev->device_features = di.features;
 
diff --git a/virtio_lo_device.h b/virtio_lo_device.h
index 2d872a7..bc40b3e 100644
--- a/virtio_lo_device.h
+++ b/virtio_lo_device.h
@@ -25,7 +25,6 @@ struct virtio_lo_device {
 	unsigned idx;
 	u32 device_id;
 	u32 vendor_id;
-	int card_index;
 
 	struct platform_device *pdev;
 
diff --git a/virtio_lo_driver.c b/virtio_lo_driver.c
index 30d1852..d7f517d 100644
--- a/virtio_lo_driver.c
+++ b/virtio_lo_driver.c
@@ -284,9 +284,6 @@ static int virtio_lo_probe(struct platform_device *pdev)
 	vl_driv->device = device;
 	vl_driv->pdev = pdev;
 
-#ifdef CONFIG_VIRTIO_LO_DEVICE_INDEX
-	vl_driv->vdev.card_index = device->card_index;
-#endif /* CONFIG_VIRTIO_LO_DEVICE_INDEX */
 	vl_driv->vdev.dev.parent = &pdev->dev;
 	vl_driv->vdev.dev.release = virtio_lo_release_dev_empty;
 	vl_driv->vdev.config = &virtio_lo_config_ops;
-- 
2.25.1

